MAUI/DataForm/localization.md

```diff
@@ -0,0 +1,182 @@
+
+## Localizing data form display values
+
+Here, the display attributes or data form item display values get localized based on culture from Localization Resource File (.Resx).
+
+#### Using attribute
+
+`ResourceType` property of [Display](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.displayattribute?view=net-7.0) attribute specifies the Resources File (.Resx) which is used to localize the values of `Name`, `ShortName`, `GroupName`, `Prompt` and `ItemsSource`.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(Name = "First Name", ShortName = "Name", GroupName = "Details", Prompt = "Type first name", ResourceType = typeof(DataFormLocalization))]
+public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can also localize the DataFormItem's `LabelText`, `PlaceHolderText` and `GroupName` in the `GenerateDataFormItem` event of `SfDataForm` by using the Resources (.Resx) file
+
+Here, the string member of .resx file will be accessed through the class (in resxFilename.Designer.cs) which will be auto-generated when .resx file is created and static string members get localized using [ResourceManager](https://learn.microsoft.com/en-us/dotnet/api/system.resources.resourcemanager.getstring?view=net-7.0) based on culture.
+
+{% tabs %}
+{% highlight C# %}
+
+[Display(Name = "First Name", GroupName = "Details", Prompt = "Type first name")]
+public String FirstName { get; set; }
+
+this.dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+{
+    if (e.DataFormItem?.GroupName == "Details")
+    {
+        e.DataFormItem.GroupName = DataFormLocalization.GroupName;
+    }
+    if (e.DataFormItem.LabelText == "First Name")
+    {
+        e.DataFormItem.LabelText = DataFormLocalization.FirstName;
+        e.DataFormItem.PlaceholderText = DataFormLocalization.PlaceHolder;
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## Localizing validation error messages
+
+Here, the `ValidationAttributes` or data form error messages get localized based on culture from Localization Resource File (.Resx).
+
+#### Using attribute
+
+The `ValidationAttribute` error message can be localized using [ErrorMessageResourceType](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) and [ErrorMessageResourceName](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) properties which are used to get a localized error messages from Localization Resource File (.Resx) based on culture.
```

Comment:
The `ValidationAttribute` error message can be localized using the [ErrorMessageResourceType](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) and [ErrorMessageResourceName](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.errormessageresourcetype?redirectedfrom=MSDN&view=net-5.0#System_ComponentModel_DataAnnotations_ValidationAttribute_ErrorMessageResourceType ) properties, which are used to get localized error messages from the Localization Resource File (.Resx) based on culture.